main.go

```diff
@@ -3,21 +3,73 @@ package main
 import (
 	"os"
 
+	"strconv"
+	"time"
+
 	"github.com/Sirupsen/logrus"
+	"github.com/docker/docker/client"
+	"github.com/rancher/gc-manager/imagegc"
 	"github.com/urfave/cli"
 )
 
 var VERSION = "v0.0.0-dev"
 
+const (
+	HighThreshold = "IMAGE_GC_HIGH_THRESHOLD"
+	LowThreshold  = "IMAGE_GC_LOW_THRESHOLD"
+	MinAge        = "IMAGE_GC_MIN_AGE"
+	GCInterval    = "IMAGE_GC_INTERVAL"
+)
+
 func main() {
 	app := cli.NewApp()
 	app.Name = "gc-manager"
 	app.Version = VERSION
 	app.Usage = "You need help!"
-	app.Action = func(c *cli.Context) error {
-		logrus.Info("I'm a turkey")
-		return nil
-	}
-
+	app.Action = run
 	app.Run(os.Args)
 }
+
+func run(c *cli.Context) error {
+	dockerclient, err := client.NewEnvClient()
+	if err != nil {
+		return err
+	}
+	highThresholdPercent, err := strconv.ParseFloat(os.Getenv(HighThreshold), 64)
+	if err != nil {
+		highThresholdPercent = 90.0
+	}
+	lowThresholdPercent, err := strconv.ParseFloat(os.Getenv(LowThreshold), 64)
+	if err != nil {
+		lowThresholdPercent = 80.0
+	}
+	minAge, err := strconv.Atoi(os.Getenv(MinAge))
+	if err != nil {
+		minAge = 5
+	}
+	interval, err := strconv.Atoi(os.Getenv(GCInterval))
+	if err != nil {
+		interval = 5
+	}
+	policy := &imagegc.GCPolicy{
+		HighThresholdPercent: highThresholdPercent,
+		LowThresholdPercent:  lowThresholdPercent,
+		MinAge:               time.Minute * time.Duration(minAge),
+		GCInterval:           time.Minute * time.Duration(interval),
+	}
+	gcManager, err := imagegc.NewImageManagerImpl(policy, dockerclient)
+	if err != nil {
+		return err
+	}
+	gcManager.Start()
```

Comment:
Actually, I see why now: Start() needs to be running on a separate interval so that it can start building a history of images and their last used time